testing/integration/tests/main.rs

```diff
@@ -1526,6 +1529,121 @@ fn test_readonly_txn_set_integer() {
+        StateTreeVersion::V5,
+        MemoryBlockstore::default(),
+    )
+    .unwrap();
+
+    let (id, sender) = tester.create_account().unwrap();
+    tester
+        .state_tree
+        .as_mut()
+        .unwrap()
+        .mutate_actor(id, |s| {
+            s.deposit_funds(&TokenAmount::from_whole(10)).unwrap();
+            Ok(())
+        })
+        .unwrap();
+
+    // Send to an f4 to create a placeholder. Otherwise, we end up invoking a constructor.
+    let receiver = Address::new_delegated(10, b"foobar").expect("failed to construct f4 address");
+
+    tester.instantiate_machine(DummyExterns).unwrap();
+
+    let executor = tester.executor.as_mut().unwrap();
+    let amount = TokenAmount::from_atto(10);
+
+    let message = Message {
+        from: sender,
+        to: receiver,
+        gas_limit: 10000000,
+        method_num: METHOD_SEND,
+        sequence: 0,
+        value: amount.clone(),
+        gas_fee_cap: TokenAmount::from_atto(1),
+        gas_premium: TokenAmount::from_atto(1),
+        ..Message::default()
+    };
+
+    let sender_pre_balance = executor
+        .state_tree()
+        .get_actor_by_address(&sender)
+        .unwrap()
+        .unwrap()
+        .balance;
+
+    let res = executor
+        .execute_message(message, ApplyKind::Explicit, 100)
+        .unwrap();
+    let sender_post_balance = executor
+        .state_tree()
+        .get_actor_by_address(&sender)
+        .unwrap()
+        .unwrap()
+        .balance;
+    assert!(res.msg_receipt.exit_code.is_success());
+    assert!(
+        sender_post_balance < sender_pre_balance - amount.clone(),
+        "txn gas should be deducted"
+    );
+
+    // now we make sure no txn gas is consumed from the sender
+    // only the funds are transferred
+
+    let message = Message {
+        from: sender,
+        to: receiver,
+        gas_limit: 10000000,
+        method_num: METHOD_SEND,
+        sequence: 1,
+        value: amount.clone(),
+        gas_fee_cap: TokenAmount::from_atto(1),
+        gas_premium: TokenAmount::from_atto(1),
+        ..Message::default()
+    };
+
+    let sender_pre_balance = executor
+        .state_tree()
+        .get_actor_by_address(&sender)
+        .unwrap()
+        .unwrap()
+        .balance;
+
+    let f = |sender: ActorID, _: &Receipt, gas: &GasOutputs| {
+        let GasOutputs {
+            base_fee_burn,
+            over_estimation_burn,
+            miner_tip,
+            refund,
+            ..
+        } = gas;
+        let total = base_fee_burn + over_estimation_burn + refund + miner_tip;
+        vec![(sender, total)]
+    };
+    let options = ExecutionOptions {
+        always_revert: false,
+        txn_gas_hook: f,
+    };
+
+    let res = executor
+        .execute_message_with_options(message, ApplyKind::Explicit, 100, options)
+        .unwrap();
+
+    let sender_post_balance = executor
+        .state_tree()
+        .get_actor_by_address(&sender)
+        .unwrap()
```

Comment:
nit: I recommend to use one `.unwrap()` per line AND binding. That guarantees unique stacktraces.

Reply:
updated!